### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by misha200119**
[DEMO LINK](https://misha200119.github.io/react_people-table-advanced/)

### CODE REVIEW

**Review by Anastasiia-Svintsova — APPROVED** [review]
Good

File: src/components/Header/Header.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//       background-color: rgb(55, 55, 55);
+//     }
+//   }
+
+//   &__link {
+//     font-size: 30px;
+//     text-decoration: none;
+//     color: gray;
+//   }
+// }
```
Comment (MaksimusCRS) [inline src/components/Header/Header.scss:24]: remove it

File: src/components/PeoplePage/PeoplePage.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    setPeople(peopleWithParents);
+    peopleToShow = peopleWithParents;
+  }, []);
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  const applyQuery = useCallback(
+    debounce((newQuery: string) => {
```
Comment (MaksimusCRS) [inline src/components/PeoplePage/PeoplePage.tsx:47]: 🔥

File: src/components/PeopleTable/PeopleTable.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      searchParams.set('sortOrder', 'desc');
+    } else {
+      searchParams.delete('sortBy');
+      searchParams.delete('sortOrder');
+    }
+
+    navigate(`?${searchParams.toString()}`);
+  }, [searchParams]);
+
+  // console.log('rerender');
```
Comment (MaksimusCRS) [inline src/components/PeopleTable/PeopleTable.tsx:48]: Remove comments before creating pr

File: src/components/PeopleTable/PeopleTable.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+              <th
+                className={classNames(
+                  'PeopleTable__column',
+                  {
+                    'PeopleTable__now-sorting-column':
+                    title === currentSortBy,
+                    'PeopleTable__now-sorting-column--sort-asc':
+                    currentSortOrder === 'asc' && title === currentSortBy,
+                    'PeopleTable__now-sorting-column--sort-desc':
+                    currentSortOrder === 'desc' && title === currentSortBy,
```
Comment (MaksimusCRS) [inline src/components/PeopleTable/PeopleTable.tsx:68]: u can use ternary operator here

File: src/functions/generateCallbackSort.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    } else if (fieldName === 'father') {
+      f1 = h1.fatherName;
+      f2 = h2.fatherName;
+    }
+
+    if (f1 === f2) {
+      return 0;
+    }
+
+    if (f1 === null) {
```
Comment (MaksimusCRS) [inline src/functions/generateCallbackSort.ts:27]:
```suggestion
    if (!f1) {
```

File: src/functions/generateCallbackSort.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      return 1;
+    }
+
+    if (f2 === null) {
```
Comment (MaksimusCRS) [inline src/functions/generateCallbackSort.ts:31]:
```suggestion
    if (!f2) {
```

File: src/functions/generateCallbackSort.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  const multiplier = sortOrder === 'asc' ? 1 : -1;
+  let out: number;
+
+  const fieldName = sortBy.toLowerCase();
+
+  // i don't know how set sortBy str as keyof
+  // HumanWithParents because i take that value from table headings
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const callback = (h1: any, h2: any) => {
```
Comment (MaksimusCRS) [inline src/functions/generateCallbackSort.ts:11]:
```suggestion
  const callback = (h1, h2) => {
```
Why any type?

File: src/functions/generateCallbackSort.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
+export const generateCallbackSort = (sortBy: string, sortOrder: string) => {
+  const multiplier = sortOrder === 'asc' ? 1 : -1;
+  let out: number;
+
+  const fieldName = sortBy.toLowerCase();
+
+  // i don't know how set sortBy str as keyof
+  // HumanWithParents because i take that value from table headings
```
Comment (MaksimusCRS) [inline src/functions/generateCallbackSort.ts:8]: U can modify headings or create enum with heading names

**Review by MaksimusCRS — APPROVED** [review]
GJ

File: src/functions/generateCallbackSort.ts (hunk 8 above)
Comment (misha200119) [reply src/functions/generateCallbackSort.ts:8]: and after event catch value and thhrow it in action like 'some_string as EnumHeadings'

**Pull request closed without merging**